### DISCUSSION TITLE
GetResponseBodyAsync not working (from DevtoolsClient's Network.ResponseReceived callback)

### DISCUSSION THREAD

**Discussion by DevCyber23** [cefsharp/CefSharp#4767] [Q&A]
Hello!

I'm trying to use GetResponseBodyAsync (from DevtoolsClient's Network.ResponseReceived callback) to get the full body of a HTTP response, however this API never returns. I call it in a synchronous way, like this: 
`var testBody = devToolsClient.Network.GetResponseBodyAsync(args.RequestId).Result; 
`However this never returns. I was expecting the function to return the body data (even if not immediately), but it never works for me. Is there a specific way I should call this API? Or maybe a setup I need to do when instantiating the Devtools?
If not, is this a bug? Can it be fixed?

Thanks!

**Comment by amaitland** [comment]
> Is there a specific way I should call this API?

The API must be called in an asynchronous fashion. 

> is this a bug

Highly unlikely.

You appear to be blocking the thread used to process responses.
